### DISCUSSION TITLE
wantsJson

### DISCUSSION THREAD

**Discussion by robert-abram** [spatie/laravel-data#105] [Q&A]
Helo,
Is there equvalent of Request wantsJson function in laravel-data ?

**Comment by rubenvanassche** [comment]
Hi @abrampl,

What would be the exact use case for such a method?

**Reply by robert-abram** [reply]
Hi,
I'm using Pipeline and depends on request i need to set proper response or redirect. In my case of course i want to use Laravel Data as LoginRequest.


In fact, this trait is something that give me info, what type of Request is it. 

```
Illuminate\Http\Concerns\InteractsWithContentTypes
```

```php
return (new Pipeline(app()))->send($request)->through([
            EnsureLoginIsNotThrottled::class,
            RedirectTwoFactorAuthenticatable::class,
            AttemptToAuthenticate::class,
            PrepareAuthenticatedSession::class,
        ])->then(function (LoginRequest $request) {
            return $request->wantsJson()
                ? response()->json(['two_factor' => false])
                : redirect()->intended(RouteServiceProvider::HOME);
        });
```

**Reply by rubenvanassche** [reply]
Why not inject both the request and data? Then you can call `wantsJson` on the request and return the data object. 

The data object is not a request, it only will be populated with the attributes of the request. 

Returning a data object works like an API resource, when the requests wants JSON a JSON version of the data object will be returned.

So I guess that will solve your problem?
